Make audio file processing more efficient

yenaing-oo opened the pull request:
Skips usage of `ffmpeg` if all audio file criteria is already met.

yenaing-oo committed:
Make audio processing more efficient, fix variable casing

yenaing-oo commented:
@danloveg 
Here are some test files I made for this feature:
[mp3_test_files.zip](https://github.com/user-attachments/files/18772052/mp3_test_files.zip)

yenaing-oo commented on lib/model/QubitDigitalObject.php:
```diff
@@ -2562,21 +2562,31 @@ public static function convertAudioToMp3($originalPath, $newPath)
-                if (strpos($output[$i], 'output buffer too small')) {
-                    $error = false;
+        $reformatFile = $formatName !== 'mp3' ||
+                        strtolower(pathinfo($originalPath, PATHINFO_EXTENSION)) !== 'mp3';
+        
+        error_log("Format name: ".$formatName);
 
-                    break;
-                }
-            }
```
I don't think the original code block here was doing anything useful, so I removed it.

danloveg replied:
You're right, it wasn't doing anything. I do have a concern that they were doing some post-ffmpeg error checking though, see my other comment.

danloveg commented on lib/model/QubitDigitalObject.php:
```diff
@@ -2562,21 +2562,31 @@ public static function convertAudioToMp3($originalPath, $newPath)
+            $status = copy($originalPath, $newPath) ? 0 : 1;
         }
 
+        if ($status !== 0) {
+            return false;
+        }  
```
I'm worried that they were doing some checking in that bit of code you removed for whether a non-zero exit code was actually a non-issue.

This might cover our bases better:

```suggestion
        if ($status !== 0 && !file_exists($newPath)) {
            return false;
        }  
```

yenaing-oo replied:
This looks good to me.

danloveg reviewed (changes requested):
This all looks good to me, the audio is a much simpler case than the video. Once you've had a look at my message re: the code you removed, and the error_logs are removed, I think we're good to merge this into my branch.

yenaing-oo committed:
Apply suggestion

Co-authored-by: Daniel Lovegrove <d.lovegrove11@gmail.com>

yenaing-oo committed:
Remove error_logs

yenaing-oo committed:
Merge branch '1912-make-audio-conversion-efficient' of https://github.com/danloveg/atom into 1912-make-audio-conversion-efficient

yenaing-oo commented:
> This all looks good to me, the audio is a much simpler case than the video. Once you've had a look at my message re: the code you removed, and the error_logs are removed, I think we're good to merge this into my branch.

I've applied your suggested change regarding the error-checking, and removed the error_logs.
Addressed by: https://github.com/danloveg/atom/pull/2/commits/a46620cf0512b85e5d4cb90baddb8a2cafc25dfc and https://github.com/danloveg/atom/pull/2/commits/be4715a324518af303a06ef74e37772aaab16f2d
